causalml/inference/tree/uplift.pyx

```diff
@@ -297,22 +314,44 @@ class UpliftTreeClassifier:
 
         self.feature_imp_dict = defaultdict(float)
 
-        if self.evaluationFunction == self.evaluate_DDP and self.n_class > 2:
-            raise ValueError("The DDP approach can only cope with two class problems, that is two different treatment "
+        if self.n_class > 2 and (self.evaluationFunction == self.evaluate_DDP or self.evaluationFunction == self.evaluate_IDDP or
+                                 self.evaluationFunction == self.evaluate_IT or self.evaluationFunction == self.evaluate_CIT):
+            raise ValueError("The DDP, IDDP, IT, and CIT approach can only cope with two class problems, that is two different treatment "
                              "options (e.g., control vs treatment). Please select another approach or only use a "
                              "dataset which employs two treatment options.")
 
+        if self.honesty:
+            try:
+                X, X_est, treatment_idx, treatment_idx_est, y, y_est = sklearn.model_selection.train_test_split(X,
+                                                                                                                treatment_idx,
+                                                                                                                y,
+                                                                                                                stratify=[treatment_idx, y],
+                                                                                                                test_size=self.estimation_sample_size,
+                                                                                                                shuffle=True,
+                                                                                                                random_state=self.random_state)
+            except ValueError:
```

Comment:
Silent fallback is not good because we wouldn't know which code is executed. Let's either (1) make stratified sampling optional and fail if there's `ValueError`, or (2) provide a log message when stratified sampling fails and non-stratified sampling is used as a fallback.

Reply:
I feel this point can be addressed in a separate PR. I will approve this PR.